pkgs/development/tools/parsing/tree-sitter/update_impl.py

```diff
@@ -14,13 +15,31 @@
 mode: str = sys.argv[1]
 jsonArg: dict = json.loads(sys.argv[2])
 
-Args = Iterator[str]
+Args = Iterator[str] | [str]
 
 
 def log(msg: str) -> None:
     print(msg, file=sys.stderr)
 
 
+def check_grammar(data) -> bool:
+    path = data["path"]
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # Copy grammar to a temporary directory
+        run_cmd(["cp", "--no-preserve=mode,ownership", "-r", f"{path}/.", tmpdirname])
+
+        # Run the tree-sitter test command
+        test_result = sub.run(
+            [bins["tree-sitter"], "test"],
```

Comment:
Shouldn't we do such checks in `checkPhase`?

Reply:
We do above https://github.com/NixOS/nixpkgs/pull/391520/files#diff-9864f57536a58149343f51f3cf58a239e47aa779966cff69917a8675df44eaa3R46-R52

Reply:
This is for marking a grammar as broken during the updater

Reply:
Oh, right, sorry